### PR TITLE
More refactorings.

### PR DESCRIPTION
**Pull request by vdbergh**
Get rid of all calls to worker_exit().

Eliminate global variable "config".

Move all parsing of parameters to setup_config_file().

Rename main() to worker().

Refactor adjust_tc.

### CODE REVIEW

**Comment by vdbergh** [comment]
Ok I see it I think.

**Comment by vdbergh** [comment]
3 cpus is the default config option. I set it to 1 now. But maybe it is better to set the default to max(1,multiprocessing.cpu_count()-1)? What do you think?

**Comment by vdbergh** [comment]
Or `min(3,max(1,multiprocessing.cpu_count()-1))`.

**Comment by ppigazzini** [comment]
I agree.

File: worker/worker.py [hunk 1]
```diff
@@ -196,23 +359,32 @@ def fetch_and_handle_task(worker_info, password, remote, current_state):
...
 
-    success = True
     run, task_id = req["run"], req["task_id"]
     current_state["run"] = run
     current_state["task_id"] = task_id
+
+    success = False
     try:
         pgn_file = run_games(worker_info, password, remote, run, task_id)
+        succes = True
```
Comment (ppigazzini) [inline worker/worker.py:377]: Typo

File: worker/worker.py [hunk 2]
```diff
@@ -340,115 +460,11 @@ def main():
...
-
-    with open(config_file, "w") as f:
-        config.write(f)
-    if options.only_config == "True":
-        worker_exit(0)
-    # We are now finished with handling command line parameters and the config file
+    options = setup_parameters(config_file)
+    if options is None:
+        return 1
+    if options.only_config==True:
```
[the remark this answers is not in the document]
Comment (vdbergh) [reply worker/worker.py:466]: Yep :(

**Comment by ppigazzini** [comment]
Started a couple of workers on DEV.

**Comment by vdbergh** [comment]
I guess strictly speaking, `fish.exit` should also have return code `0`. I think this was the old behaviour since in that case `worker_exit()` wasn't called.

File: worker/worker.py [hunk 3]
```diff
@@ -522,8 +539,10 @@ def main():
...
     print("Waiting for the heartbeat thread to finish...")
     heartbeat_thread.join()
-    print("Finishing the worker normally")
-
+    
+    return 0 if fish_exit else 1
 
 if __name__ == "__main__":
-    main()
+    ret=worker()
```
Comment (ppigazzini) [inline worker/worker.py:546]: If I'm not wrong `sys.exit(worker())` should made the work, see
https://softwareengineering.stackexchange.com/questions/418600/best-practice-for-python-main-function-definition-and-program-start-exit

**Comment by vdbergh** [comment]
I mean if something goes wrong.

**Comment by ppigazzini** [comment]
We had so many naked except :(

I'm wondering if we can retire:
```python
def worker_exit(val=1):
    os._exit(val)
```

**Comment by ppigazzini** [comment]
`sys.exit()` is the standard way to exit, I'm to retire `worker_exit()`

**Comment by vdbergh** [comment]
Calling it a day now.

**Review by ppigazzini — APPROVED** [review]

**Pull request merged**